add new features and fix some bugs

ykwd opened the pull request:
1. add feature : deploy service on windows cluster via app store; 
2. modify machine_pool_mgr's get_machine method to allow more flexible options. 
3. change the line ending of mahcine_pool_mgr.cpp/.h from CR LF to LF

imzhenyu commented:
Thanks, @ykwd. No worry, I can help do that later. Happy Holiday~

ykwd commented:
Thank you, @imzhenyu . I hope that is just caused by my carelessness.

qinzuoyan commented:
It is not the tab problem, it's the "<::dsn" problem. Now it's not allowed to contain "<::", please replace to "< ::dsn".

imzhenyu commented:
Thx Zuoyan. @ykwd now you know it is not the tab's fault:-)

ykwd commented:
thanks. @qinzuoyan . It's very appreciate that you can discover my mistake. It's glad to see this pr can finally pass the test.

ykwd commented:
Maybe this fault can be avoided if I have merged to the newest version before the pr.

imzhenyu commented on scripts/windows/deploy.one.cmd:
```diff
@@ -34,8 +34,6 @@ REM
     set rdst=\\%machine%\%rdst_dir%
     @mkdir %rdst%    
     xcopy /F /Y /S %src_dir% %rdst%
-    COPY /Y %bin_dir%\7z.exe %rdst%
```
not sure you can delete this when we have monitor package deployed together.

ykwd replied:
The monitor will install 7z itself. And I thought this line is used when we unziped the package on the remote machine while we now unzip the package locally.

imzhenyu replied:
there is chick-and-egg issue. The monitor's pack needs 7z to unzip itself, after that the monitor pack may install 7z itself (i'm not sure the later stuff though)

ykwd replied:
does the code to unzip the package exist in this script?

imzhenyu replied:
it is in the publish script - but you are right, should 7z be copied in publish script instead.

imzhenyu commented on scripts/windows/deploy.cmd:
```diff
@@ -48,7 +48,7 @@ IF "%3" EQU "" (
 ::SET rdst_dir=%ldst_dir::=$%
 ::SET machine=%5
-FOR /F %%i IN (%machine_list%) DO ECHO %cmd% %%i ... && start %bin_dir%\deploy.one.cmd %cmd% %src_dir% %ldst_dir% %deploy_name% %%i
+FOR /F %%i IN (%machine_list%) DO ECHO %cmd% %%i ... && CALL %bin_dir%\deploy.one.cmd %cmd% %src_dir% %ldst_dir% %deploy_name% %%i
```
we may leave this script unchanged and get one new copy with CALL in deploy service, to avoid reducing the deployment speed when we use the scripts directly

ykwd replied:
ok. I think so.

ykwd committed:
1. add feature : deploy service on windows cluster via app store;
2. modify machine_pool_mgr's get_machine method to allow more flexible options.
3. change the line ending of mahcine_pool_mgr.cpp/.h from CR LF to LF

ykwd commented:
I create a new copy of deploy.cmd and deploy.one.cmd, and leave the original scripts unchanged.

imzhenyu commented:
Thanks @ykwd Happy Chinese New Year~

ykwd commented:
happy new year! ≧^◡^≦
